### ISSUE TITLE
Add custom pathing for mobs, Potentially add block placing too?

### ISSUE THREAD

**Issue by SM9CC** [Johni0702/BetterPortals#63]
So first off, this mod is mindblowing, the work you have done so far is amazing.

I had a couple of ideas (Should be possible I think)

- Add custom pathing task so mobs can follow you through the portal?
- Add ability to place blocks, a good example would be a rail system which travels though portals seamlessly.

Thanks!

**Comment by Johni0702** [comment]
It may be possibly to modify some of the AI tasks to work with portals but it may be quite tricky to do. And I might also be missing some important detail because I'm not at all familiar with it and only just now had a look.
I've keeping this issue open for that feature.

For placing blocks in portals, there's #44 but I've got really no good idea to solve it (especially no generic way). The best thing I can think of is making portal blocks tile entities (possibly on demand) and then more or less re-implementing whichever thing you want to be able to work between the two dimensions.

For rail transport there's also the issue that entities can't use portals when they're riding/being ridden, neither in vanilla MC nor (atm) with BP.

**Comment by SM9CC** [comment]
Understandably tricky so no rush :) 

Thanks.

**Comment by DarianLStephens** [comment]
How about just placing blocks on the opposite side of the portal? Say you want to place a block in the Nether while in the Overworld, that would be good.
At the very least, it'd be nice to get rid of the bounding box on the portal blocks.

**Comment by SM9CC** [comment]
> How about just placing blocks on the opposite side of the portal? Say you want to place a block in the Nether while in the Overworld, no connection between them. That would be good.

That sounds a lot easier.

**Comment by Johni0702** [comment]
> How about just placing blocks on the opposite side of the portal? Say you want to place a block in the Nether while in the Overworld, no connection between them. That would be good.


https://github.com/Johni0702/BetterPortals/issues/10
Gets surprisingly (or unsurprisingly if you're used to portals) tricky to implement though once you consider that stuff which gets placed / broken can depend on the acting entity's position (e.g. piston placement) and potentially other properties.

> At the very least, it'd be nice to get rid of the bounding box on the portal blocks.

Not sure what you mean. There is no bounding box on the portal blocks otherwise you'd see their outlines when you look at them (well, they have one but it's of zero size, so they effectively have none).

Also, I'd prefer to keep this issue on custom AI pathing only. Please open a separate issue for other things.

**Comment by DarianLStephens** [comment]
Right, I've just discovered that the blocks don't have a bounding box; it's blocks directly against the back of the portal whose bounding box is poking through the screen.

**Comment by Johni0702** [comment]
> Right, I've just discovered that the blocks don't have a bounding box; it's blocks directly against the back of the portal whose bounding box is poking through the screen.

Good catch. MC draws them with a tiny (but non-zero) offset from the block which is why they show through. Fixing #10 requires fixing the code that checks which block you're looking at, which will also then fix this rendering issue.
